### ISSUE TITLE
Crash in GetLoginItemSettings on macOS 10.13

### ISSUE THREAD

**Issue by CharlieHess** [electron/electron#10561]
* Electron version: `1.7.7`, `1.6.12`
* Operating system: macOS 10.13

We're getting several reports of a native crash on startup for the Slack app, when running the High Sierra beta. One in particular also tried the Trello Electron app and found it crashed in the same way:

>Yesterday I install the Trello app from the Mac App Store and it acts like the Slack app, I’m getting the same error from Apple. Something with com.github.electron.framework at GetLoginItemSettings() function.

<details>

```
Process: Slack [21601] 
Path: /Applications/Slack.app/Contents/MacOS/Slack 
Identifier: com.tinyspeck.slackmacgap 
Version: 2.8.0 (4320) 
Code Type: X86-64 (Native) 
Parent Process: ??? [1] 
Responsible: Slack [21601] 
User ID: 501

Date/Time: 2017-09-18 23:43:10.229 +0200 
OS Version: Mac OS X 10.13 (17A360a) 
Report Version: 12 
Anonymous UUID: BE4E1D94-B2A3-7F73-5F03-897E8B01F443

Sleep/Wake UUID: 03C2BD4F-42DF-48E7-BB65-85747C3FA58B

Time Awake Since Boot: 93000 seconds 
Time Since Wake: 17000 seconds

System Integrity Protection: enabled

Crashed Thread: 0 CrBrowserMain Dispatch queue: com.apple.main-thread

Exception Type: EXC_BAD_INSTRUCTION (SIGILL) 
Exception Codes: 0x0000000000000001, 0x0000000000000000

Application Specific Information: 
*** CFEqual() called with NULL first argument ***

Thread 0 Crashed:: CrBrowserMain Dispatch queue: com.apple.main-thread 
0 com.apple.CoreFoundation 0x00007fff3d06e73c CFEqual + 620 
1 com.github.electron.framework 0x000000010d761856 0x10d53a000 + 2259030 
2 com.github.electron.framework 0x000000010d7616f2 0x10d53a000 + 2258674 
3 com.github.electron.framework 0x000000010d5d61e8 atom::Browser::GetLoginItemSettings(atom::Browser::LoginItemSettings const&) + 24 
4 com.github.electron.framework 0x000000010d54688d atom::api::App::GetLoginItemSettings(mate::Arguments*) + 61 
5 com.github.electron.framework 0x000000010d5512a9 0x10d53a000 + 94889 
6 com.github.electron.framework 0x000000010d55126e 0x10d53a000 + 94830 
7 com.github.electron.framework 0x000000010d551244 0x10d53a000 + 94788 
8 com.github.electron.framework 0x000000010d55120f base::internal::Invoker<base::internal::BindState<atom::Browser::LoginItemSettings (atom::api::App::*)(mate::Arguments*)>, atom::Browser::LoginItemSettings (atom::api::App*, mate::Arguments*)>::Run(base::internal::BindStateBase*, atom::api::App*&&, mate::Arguments*&&) + 31 
9 com.github.electron.framework 0x000000010d550f8f 0x10d53a000 + 94095 
10 com.github.electron.framework 0x000000010d550c99 0x10d53a000 + 93337 
11 com.github.electron.framework 0x000000010d550b4c mate::internal::Dispatcher<atom::Browser::LoginItemSettings (atom::api::App*, mate::Arguments*)>::DispatchToCallback(v8::FunctionCallbackInfo<v8::Value> const&) + 124 
12 libnode.dylib 0x0000000111a00d55 0x111805000 + 2080085 
13 libnode.dylib 0x0000000111ab1d18 0x111805000 + 2805016 
14 libnode.dylib 0x0000000111ab1271 0x111805000 + 2802289 
15 ??? 0x000027bbb9e843a7 0 + 43687231374247 
16 ??? 0x000027bbba429d8e 0 + 43687237295502 
17 ??? 0x000027bbba429a84 0 + 43687237294724 
18 ??? 0x000027bbb9e85cd5 0 + 43687231380693 
19 ??? 0x000027bbb9f0915a 0 + 43687231918426 
20 ??? 0x000027bbb9ef23f8 0 + 43687231824888 
21 ??? 0x000027bbba3a3ff2 0 + 43687236747250 
22 ??? 0x000027bbba3a3dea 0 + 43687236746730 
23 ??? 0x000027bbba2f957b 0 + 43687236048251 
24 ??? 0x000027bbba429822 0 + 43687237294114 
25 ??? 0x000027bbba2f957b 0 + 43687236048251 
26 ??? 0x000027bbba4042ea 0 + 43687237141226 
27 ??? 0x000027bbba40419c 0 + 43687237140892 
28 ??? 0x000027bbba2f957b 0 + 43687236048251 
29 ??? 0x000027bbba42962b 0 + 43687237293611 
30 ??? 0x000027bbba4286c4 0 + 43687237289668 
31 ??? 0x000027bbb9e85cd5 0 + 43687231380693 
32 ??? 0x000027bbb9ef1ac3 0 + 43687231822531 
33 ??? 0x000027bbb9eb65c1 0 + 43687231579585 
34 libnode.dylib 0x0000000111c7e4a5 0x111805000 + 4691109 
35 libnode.dylib 0x0000000111c7e209 v8::internal::Execution::Call(v8::internal::Isolate*, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, int, v8::internal::Handle<v8::internal::Object>*) + 169 
36 libnode.dylib 0x0000000111a1fd5b v8::Function::Call(v8::Local<v8::Context>, v8::Local<v8::Value>, int, v8::Local<v8::Value>*) + 459 
37 libnode.dylib 0x00000001118df8f6 node::MakeCallback(node::Environment*, v8::Local<v8::Value>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*) + 726 
38 libnode.dylib 0x00000001118dfe58 node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*) + 104 
39 libnode.dylib 0x00000001118dfdbf node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*) + 95 
40 libnode.dylib 0x00000001118dfd30 node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, char const*, int, v8::Local<v8::Value>*) + 80 
41 com.github.electron.framework 0x000000010d63375a mate::internal::CallMethodWithArgs(v8::Isolate*, v8::Local<v8::Object>, char const*, std::__1::vector<v8::Local<v8::Value>, std::__1::allocator<v8::Local<v8::Value> > >*) + 74 
42 com.github.electron.framework 0x000000010d54c1a9 0x10d53a000 + 74153 
43 com.github.electron.framework 0x000000010d5a81ab 0x10d53a000 + 450987 
44 com.github.electron.framework 0x000000010d5a8114 0x10d53a000 + 450836 
45 com.github.electron.framework 0x000000010d59f47d 0x10d53a000 + 414845 
46 com.github.electron.framework 0x000000010d5a093e atom::api::WebContents::DocumentLoadedInFrame(content::RenderFrameHost*) + 62 
47 com.github.electron.framework 0x000000010d5a0960 non-virtual thunk to atom::api::WebContents::DocumentLoadedInFrame(content::RenderFrameHost*) + 16 
48 com.github.electron.framework 0x000000010dd4cc74 0x10d53a000 + 8465524 
49 com.github.electron.framework 0x000000010dd4ca96 0x10d53a000 + 8465046 
50 com.github.electron.framework 0x000000010dd4c146 0x10d53a000 + 8462662 
51 com.github.electron.framework 0x000000010dbbb871 0x10d53a000 + 6822001 
52 com.github.electron.framework 0x000000010e23f98b 0x10d53a000 + 13654411 
53 com.github.electron.framework 0x000000010d79b351 0x10d53a000 + 2495313 
54 com.github.electron.framework 0x000000010d76df26 0x10d53a000 + 2309926 
55 com.github.electron.framework 0x000000010d76e1fc 0x10d53a000 + 2310652 
56 com.github.electron.framework 0x000000010d76e543 0x10d53a000 + 2311491 
57 com.github.electron.framework 0x000000010d77188d 0x10d53a000 + 2324621 
58 com.github.electron.framework 0x000000010d72e73a 0x10d53a000 + 2049850 
59 com.github.electron.framework 0x000000010d771304 0x10d53a000 + 2323204 
60 com.apple.CoreFoundation 0x00007fff3d0f56c1 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17 
61 com.apple.CoreFoundation 0x00007fff3d1af2cc __CFRunLoopDoSource0 + 108 
62 com.apple.CoreFoundation 0x00007fff3d0d8160 __CFRunLoopDoSources0 + 208 
63 com.apple.CoreFoundation 0x00007fff3d0d75dd __CFRunLoopRun + 1293 
64 com.apple.CoreFoundation 0x00007fff3d0d6e43 CFRunLoopRunSpecific + 483 
65 com.apple.HIToolbox 0x00007fff3c3f6866 RunCurrentEventLoopInMode + 286 
66 com.apple.HIToolbox 0x00007fff3c3f65d6 ReceiveNextEventCommon + 613 
67 com.apple.HIToolbox 0x00007fff3c3f6354 _BlockUntilNextEventMatchingListInModeWithFilter + 64 
68 com.apple.AppKit 0x00007fff3a6f444f _DPSNextEvent + 2085 
69 com.apple.AppKit 0x00007fff3ae89508 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 3044 
70 com.apple.AppKit 0x00007fff3a6e925d -[NSApplication run] + 764 
71 com.github.electron.framework 0x000000010d7720de 0x10d53a000 + 2326750 
72 com.github.electron.framework 0x000000010d77174c 0x10d53a000 + 2324300 
73 com.github.electron.framework 0x000000010d76dc8e 0x10d53a000 + 2309262 
74 com.github.electron.framework 0x000000010d783653 0x10d53a000 + 2397779 
75 com.github.electron.framework 0x000000010d9d387b 0x10d53a000 + 4823163 
76 com.github.electron.framework 0x000000010d9d36d0 0x10d53a000 + 4822736 
77 com.github.electron.framework 0x000000010d9d5e12 0x10d53a000 + 4832786 
78 com.github.electron.framework 0x000000010d9cf7ac 0x10d53a000 + 4806572 
79 com.github.electron.framework 0x000000010d93530d 0x10d53a000 + 4174605 
80 com.github.electron.framework 0x000000010d934406 0x10d53a000 + 4170758 
81 com.github.electron.framework 0x000000010d53bd07 AtomMain + 71 
82 com.tinyspeck.slackmacgap 0x000000010cd38f26 main + 38 
83 libdyld.dylib 0x00007fff645d0145 start + 1
```

</details>

**Comment by CharlieHess** [comment]
We tried [electron-atos](https://github.com/electron/electron/files/1319463/main_out.txt) on this stack and the result is … strange. The top frame is in a `heap_profiler_heap_dump_writer`, looks like some kinda debug tracing?

**Comment by MarshallOfSound** [comment]
I feel as though this API can / should be disabled in MAS builds (It doesn't work anyway).

Is this issue also present on normal mac builds?

**Comment by CharlieHess** [comment]
@MarshallOfSound seems to happen on normal builds too.

**Comment by gerges** [comment]
Thanks for the heads up @CharlieHess / @MarshallOfSound, we'll protect this with a MacOS version check for now. You can add affects `1.6.11`, that's what Trello's using.  I'm a little baffled by this one, we've got a few internal/external users using High Sierra and haven't seen this yet.

**Comment by MarshallOfSound** [comment]
> seems to happen on normal builds too.

That makes this a bit more annoying then, afaik this API is entirely implemented in Chromium land, might have to upstream / patch a fix in

**Comment by gerges** [comment]
@MarshallOfSound I don't think this is a noop on mas like you suspected. Not only do I not see the crash on Mac OS X 10.13 (17A360a), it actually respects the hidden flag (which is what I use the call to determine) on the App Store installed version of my app.

**Comment by Thomas101** [comment]
We had some trouble with this. In our case we found it to be timing issue. Would be much worse on machines with platter disks and less than 4 gig ram but could never reproduce with electron-quick-start

We delayed the call until after the ready event fired on the app and it seemed to work fine after that

**Comment by vanshg** [comment]
Same thing is happening to me after updating to 10.13. I checked my login items and all my entries say Unknown. I am unable to delete them by pressing the minus button or removing the loginitems plist file.

It seems like multiple Electron apps are not working for me.
Slack doesn't work and produces a crash report.
Spotify also doesn't work, however it doesn't produce a crash report.

**Comment by robinwassen** [comment]
@vanshg Spotify is probably not related since it does not use Electron, they use a custom rolled Chrome Embedded Framework solution.

**Comment by indywidualny** [comment]
@vanshg The same thing just happened to me. The worst day of my life. At least I know what's wrong.

**Comment by dag777** [comment]
@vanshg I also have the same issue, sucks! Does anyone know a workaround to somehow keep using slack?

**Comment by vanshg** [comment]
If anyone's looking for a workaround, apparently Slack has an alpha where it is no longer crashing:
http://slack-ssb-updates.s3.amazonaws.com/support-test/Slack-3.0.0-alpha1709277b512d6-macOS.zip

**Comment by dag777** [comment]
YES! it works again for me too :) 👍  thanks @vanshg !

**Comment by dburr** [comment]
Thank you @vanshg , today I updated the Slack app from MAS and it stopped working for me. Deleted the MAS version and replaced it with the alpha and all is well.

I am also seeing two undeletable "Unknown" items in Login Items. So far they don't appear to be causing any trouble, but they are really bugging me. Were you able to find any way of removing them?

**Comment by vanshg** [comment]
No, unfortunately, I've still been unable to remove them. I was hoping that 10.13.1 would fix this bug, but after updating to the Public Beta, I still can't remove them

**Comment by indywidualny** [comment]
@dburr I've restored the whole machine from TM backup. Backup from the previous day the issue occurred. It fixed Slack (and Spotify but it's not probably related to the issue). For now all is just ok. So this issue is strange because it's really random. I'm still on High Sierra and have no issue now.

**Comment by kquinsland** [comment]
👋 

I'm also having this issue on osx high sierra.  Chrome, also, is refusing to load now, too.  I have not yet looked into _why_ that is, as i wanted to get this bug filed / researched first. 


Here's a partial stack trace

```
OS Version:            Mac OS X 10.13 (17A405)

Time Awake Since Boot: 450 seconds

System Integrity Protection: enabled

Crashed Thread:        0  CrBrowserMain  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_INSTRUCTION (SIGILL)
Exception Codes:       0x0000000000000001, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Illegal instruction: 4
Termination Reason:    Namespace SIGNAL, Code 0x4
Terminating Process:   exc handler [0]

Application Specific Information:
*** CFEqual() called with NULL first argument ***

Thread 0 Crashed:: CrBrowserMain  Dispatch queue: com.apple.main-thread
0   com.apple.CoreFoundation      	0x00007fff4edbc73c CFEqual + 620
1   com.github.electron.framework 	0x000000010eba9976 0x10e992000 + 2193782
2   com.github.electron.framework 	0x000000010eba9812 0x10e992000 + 2193426
3   com.github.electron.framework 	0x000000010ea2db18 atom::Browser::GetLoginItemSettings(atom::Browser::LoginItemSettings const&) + 24
4   com.github.electron.framework 	0x000000010e99ea0d atom::api::App::GetLoginItemSettings(mate::Arguments*) + 61
5   com.github.electron.framework 	0x000000010e9a9429 0x10e992000 + 95273
6   com.github.electron.framework 	0x000000010e9a93ee 0x10e992000 + 95214
7   com.github.electron.framework 	0x000000010e9a93c4 0x10e992000 + 95172
8   com.github.electron.framework 	0x000000010e9a938f base::internal::Invoker<base::internal::BindState<atom::Browser::LoginItemSettings (atom::api::App::*)(mate::Arguments*)>, atom::Browser::LoginItemSettings (atom::api::App*, mate::Arguments*)>::Run(base::internal::BindStateBase*, atom::api::App*&&, mate::Arguments*&&) + 31
9   com.github.electron.framework 	0x000000010e9a910f 0x10e992000 + 94479
10  com.github.electron.framework 	0x000000010e9a8e19 0x10e992000 + 93721
11  com.github.electron.framework 	0x000000010e9a8ccc mate::internal::Dispatcher<atom::Browser::LoginItemSettings (atom::api::App*, mate::Arguments*)>::DispatchToCallback(v8::FunctionCallbackInfo<v8::Value> const&) + 124
12  libnode.dylib                 	0x0000000112e40d55 0x112c45000 + 2080085
13  libnode.dylib                 	0x0000000112ef1d18 0x112c45000 + 2805016
14  libnode.dylib  
```

**Comment by dburr** [comment]
@kquinsland I am also seeing the Chrome crashing behavior. I suspect that it is because Chrome normally puts itself in the login items when you first install it, and so it is trying to check the login items and the "Unknown" login items causes it to crash. I also see crashing when I attempt to access the "general" preferences tab in my Viscosity VPN client -- again, I suspect that it is attempting to check Login Items to make sure it is still set as a launch at login item, and the "Unknown" login items is causing it to crash.

**Comment by kquinsland** [comment]
@dburr I was looking into something unrelated to this a few moments ago and noticed that my login items are all labeled `Unknown` with a kind of `Unknown` and the unchecked `hide` box w/ the Yellow warning icon.

I *just* updated my mac w/ the `High Sierra 10.13 Supplemental Update`. Trying to figure out if that update is what hosed my login items..

**Comment by dburr** [comment]
@kquinsland Don't think the Supplemental Update caused this, as I am definitely seeing this "unknown" login item problem and I haven't yet updated to the Supplemental Update.

**Comment by kquinsland** [comment]
@dburr Sadly, it looks like this is a bug w/ the OG high sierra.  people are reporting the issue with dates before the update was rolled out:

https://forums.macrumors.com/threads/high-sierra-deleted-login-items-list.2072980/
---
It looks like the tell is seeing lines like this in your `Console.app`:

```
### Error:18446744073709551581 File:/BuildRoot/Library/Caches/com.apple.xbs/Sources/AccountsPref/AccountsPref-285/LoginItem.m Line:68
```

I'll try to get a bug opened w/ Apple's tracking system sometime today.

**Comment by kquinsland** [comment]
I've opened a bug w/ Apple about this.
https://bugreport.apple.com/web/?problemID=34857430

**Comment by vanshg** [comment]
Thanks for opening the bug @kquinsland. Unfortunately I don't think anyone but you can access the bug report so we can't express that we're also having the same issue. 

On a related note, I get the same exact error message in `Console.app`, with the same Error number and same line number in `LoginItem.m`

**Comment by kquinsland** [comment]
@vanshg thanks for looking.  Dont know how to adjust the visibility of the bug.  Perhaps open your own?

here's what the report says:

`Login Item corrupted and can not be deleted from Users & Groups after updating to macOS High Sierra`
-----
```

Summary:
My Login Items have been corrupted; they show up as `Unknown` with a yellow explanation / warning symbol.  I am unable to delete Login Items from the System Preferences UI.  Editing `plist` files that have been previously associated with the Login Items preference pane has no effect.

Various OSX API's that attempt to read the users Login items are also failing.
A limited list of apps that are currently crashing because the API to fetch the user's login items is broken:

- Google Chrome
- Most apps based on google chrome / Github's eletron framework
- Viscosity vpn


Steps to Reproduce:

0) Be running macOS Sierra.
1) Upgrade to High Sierra


Expected Results:
I expect that my login items are not corrupted / lost / damaged.
I expect that the APIs built into OSX do not result in a crash.  Here is a truncated stack trace.  The same one (and similar ones) is available in the github issues link below.

```
OS Version:            Mac OS X 10.13 (17A405)

Time Awake Since Boot: 450 seconds

System Integrity Protection: enabled

Crashed Thread:        0  CrBrowserMain  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_INSTRUCTION (SIGILL)
Exception Codes:       0x0000000000000001, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Illegal instruction: 4
Termination Reason:    Namespace SIGNAL, Code 0x4
Terminating Process:   exc handler [0]

Application Specific Information:
*** CFEqual() called with NULL first argument ***

Thread 0 Crashed:: CrBrowserMain  Dispatch queue: com.apple.main-thread
0   com.apple.CoreFoundation      	0x00007fff4edbc73c CFEqual + 620
```


Actual Results:
When I click the `-` button on the system UI, i expect the `Unknown` entry to be removed.  It is not.  When i click the `Hide` check box, i get *another* `Unknown` entry in the list.

When I click the `-` Button in the UI, i see multiple lines in the `Console.app`.  They look like this:

```
### Error:18446744073709551581 File:/BuildRoot/Library/Caches/com.apple.xbs/Sources/AccountsPref/AccountsPref-285/LoginItem.m Line:68
```

and come from the `com.apple.preferences.users.remoteservice` process.




Version/Build:
OS Version:            Mac OS X 10.13 (17A405)


Configuration:
Here's some additional information documenting this problem:

https://github.com/electron/electron/issues/10561#issuecomment-334634029

https://discussions.apple.com/thread/8096021
https://discussions.apple.com/thread/8086931
https://discussions.apple.com/thread/8089760
```

**Comment by KevinBachus** [comment]
I opened a bug with Apple a little while ago. This behavior appears to happen in the part of several applications' code that deals with the "Start at Login" setting. The system LSSharedFileListItemResolve call returns a null URL even though it also returns a status code indicating success. The applications' code appears to assume it would be non-null and passes it as the first parameter CFEqual, which leads to a crash. This appears to have worked reliably from 10.6-10.12 but something has definitely changed in 10.13.

**Comment by bobwenx** [comment]
Maybe related topic discussed: https://discussions.apple.com/message/32379140#32379140

**Comment by CsabaZz** [comment]
@vanshg thanks for the **Slack alpha**, it really does not crash on High Sierra!

**Comment by kquinsland** [comment]
Apparently Apple is aware of this issue. My bug (34857430) was closed as a duplicate of `33524015`.  Of course, i don't have access to the other bug, it seems that appls's bug reporter is pretty locked down. :(

**Comment by vincent-paing** [comment]
The Login Items becoming unknown seems to affect Steam on my device too

**Comment by flindeberg** [comment]
So, in effect, all osx applications which use the LoginItemService (or whatever it is called) and doesn't check for null are broken? 

@vanshg  And ty for the Slack alpha!

**Comment by kquinsland** [comment]
@flindeberg, yes that's the case. 

however there is somebody that claims to have a solution.  I have not yet tested this...

```
Okay, they found the problem a couple of days ago. They know how to fix it, but as far as I know, they're not sure what caused it to happen in the first place (migration assistant is my guess).
 
Anyway, search out and destroy a file called backgrounditems.btm then reboot. The login panel should be cleared out. The senior advisor and I rebooted three times with a test app inserted through the normal login panel procedures and all was well. I added back my normal items and they'e still working just fine.
 
I'm on a new/used iMac late 2015 2.8ghz, I-5, running 10.13
 
Good luck.
```

from: https://discussions.apple.com/message/32423169?ac_cid=tw123456#32423169

**Comment by kquinsland** [comment]
It looks like the path of the `backgrounditems.btm` file is here:

```/Users/$your_user/Library/Application Support/com.apple.backgroundtaskmanagementagent/backgrounditems.btm```

found via

`sudo find / -name backgrounditems.btm -not -path '*TimeMachine*'`

**Comment by kquinsland** [comment]
Oh hell yes.  @flindeberg , the solution posted in the apple support communities _worked_ for me.

**Comment by dburr** [comment]
Can also confirm that the solution @kquinsland posted worked perfectly!!  Note that **removing that file will erase all the Login Items that are currently listed in System Preferences**, so be sure and make a note of the apps/other things that you have in there right now so that you can re-set them up, BUT it DOES get rid of all the "Unknown" items! Now all of the apps that were crashing on me (Chrome, Viscosity, Slack, etc.) are working once again.

**Comment by vanshg** [comment]
Yep, that did the trick! 
I spent a while trying to find a solution/alternate files to delete, but this file never came up. Wonder how they narrowed it down to this file as the culprit (doesn't seem like the poster on Apple's thread was an employee or anything)

**Comment by vincent-paing** [comment]
+1 for the solution @kquinsland posted

**Comment by dburr** [comment]
So today apps started crashing on me again and sure enough, those pesky Unknown items were back.

Anyway today I came up with a theory as to why this may be happening. I noticed that right before the latest unknown login items incident, several of my apps had updates that got installed. Apps that have login items associated with them. I suspect that when the apps update (i.e. get deleted and replaced by newer versions) that change is somehow not being tracked/handled properly by the login items subsystem, resulting in their login item entries changing to "unknown." I have no other evidence for this theory than my gut, and I don't really have the expertise to debug this to see if this Is actually what is happening.

**Comment by dburr** [comment]
Alright. I've finally discovered what's been causing these "unknown" login items to appear, at least for me. I am a Multiplatform mobile developer so I work with both iOS and Android devices, as such I use the [Android File Transfer](https://www.android.com/filetransfer/) app to copy data to/from my Nexus 6p. However AFT has an annoying habit of always adding itself to the Startup Items. So I used [this hack](https://gist.github.com/zeroseis/ce66d4c6b776577442a6) to prevent it from doing so. My guess is that when I run Android File Transfer, it tries to re-add its (nonexistent, since I renamed it) startup item, which causes the Unknown items to appear. So for now I've switched to using the command-line `adb` tool to copy files to/from my device, which is a bit clunky but it works. I suspect that other apps that manipulate their login items but somehow screw up in doing so is causing this for other people.

**Comment by Shchvova** [comment]
TL;DR for people who don't want to read whole bunch of stuff, what helped:
```
rm ~/Library/Application\ Support/com.apple.backgroundtaskmanagementagent/backgrounditems.btm
```

**Comment by sethwklein** [comment]
And to keep it from happening again next time you run Android File Transfer use this: https://0xstubs.org/prevent-autostart-of-android-file-transfer-on-macos/

**Comment by massi1978** [comment]
i have solution!

**Comment by massi1978** [comment]
i have solution!

**Comment by kquinsland** [comment]
@massi1978  care to do a little less double posting and a little more sharing?

**Comment by massi1978** [comment]
Sorry browser issue!

**Comment by massi1978** [comment]
The solution is:
The user account have a directory library bat is simply hidden! Open with alt or shift i dont remember and the solution is the same of version 10.13!!

**Comment by massi1978** [comment]
I hope I was helpful

**Comment by massi1978** [comment]
Sorry for my bad english!

**Comment by uerceg** [comment]
Regarding Slack crashes, reply from their support team:

```
As it turns out, we've identified a bug in this version of Slack, and the good news is that we've found a solution!

Currently, the fix is available in version 3.0.0 of Slack which is presently still in beta. The regular release will be coming soon, but if you'd like to get the fix right away, you can download the beta here: https://slack.com/beta/osx.
```

**Comment by codebytere** [comment]
We are no longer implementing bugfixes for versions of Electron <= `1.7.x`, so i'm going to close this issue but if it is still persisting in more recent versions of Electron we can certainly reopen it!
